homotopy-model/src/model/proof.rs

```diff
@@ -287,7 +287,11 @@ impl ProofState {
                     selected.boundary == boundary || globularity(&selected.diagram, &ws.diagram)
                 })
             }),
-            Action::TakeIdentityDiagram | Action::ClearWorkspace => self.workspace.is_some(),
+            Action::TakeIdentityDiagram => self
+                .workspace
+                .as_ref()
+                .map_or(false, |w| w.view.dimension < u8::MAX),
```

Comment:
Why are you checking the view dimension? The `TakeIdentityDiagram` action is valid regardless of the view dimension (it only increases the view dimension if it's 1D).

Reply:
This check actually belongs to UpdateView, you're right. It was mean to prevent overflow panics when the dimension is 255.

Reply:
Even for `UpdateView` you shouldn't need to check anything, right? You can't construct `View`s which are not between 0 and 4.

Reply:
Then I would propose to make it explicit in the "UpdateView" is_valid criterion. Right now update_view just saves whatever you pass it.